src/common/transformations/include/ov_ops/type_relaxed.hpp

```diff
@@ -283,23 +282,23 @@ bool TypeRelaxed<BaseOp>::evaluate(const HostTensorVector& outputs, const HostTe
         const auto expected_output_type = get_overridden_output_type(i);
 
         if (expected_output_type != element::undefined &&
-            original_outputs[i]->get_element_type() != expected_output_type) {
+            original_outputs[i].get_element_type() != expected_output_type) {
             if (convert == nullptr) {
                 convert = std::make_shared<ov::op::v0::Convert>();
             }
 
             convert->set_destination_type(expected_output_type);
-            const auto casted_output =
-                std::make_shared<HostTensor>(expected_output_type, original_outputs[i]->get_shape());
-            if (!convert->evaluate({outputs[i]}, {original_outputs[i]})) {
+            const auto casted_output = ov::Tensor(expected_output_type, original_outputs[i].get_shape());
+            ov::TensorVector outs = {outputs[i]};
+            ov::TensorVector ins = {original_outputs[i]};
+            if (!convert->evaluate(outs, ins)) {
```

Comment:
why is it not possible to `evaluate( {outputs[i]}, {original_outputs[i]})` as before?
why do we need non-const reference in new API?

Reply:
`evaluate( outs, {original_outputs[i]})`
This is possible as inputs are `const TensorVector&` and it can bind to rvalue. As outputs are not const reference it cannot be bind. It would required make overload evaluate which accepts `(TensorVector&& outputs,...`